src/conv-core/conv-conds.C

```diff
@@ -401,26 +247,23 @@ static void ccd_heap_remove(void)
  */
 static void ccd_heap_update(double curWallTime)
 {
-  ccd_heap_elem *h = CpvAccess(ccd_heap);
-  ccd_heap_elem *e = h+CpvAccess(ccd_heapmaxlen);
-  int i,ne=0;
+  auto & h = CpvAccess(ccd_heap);
+
+  std::vector<ccd_periodic_callback> expired;
```

Comment:
Thinking about this more, we might want to hoist this vector to persistent state in order to avoid small heap allocations everytime we pull something out of the heap. Could be in a separate commit.

Reply:
Despite the comment in this function, I don't see why we need to process the heap elements in two phases if handled correctly. Let's see if CI validates my thinking.